### DISCUSSION TITLE
Tomcat support

### DISCUSSION THREAD

**Discussion by onknows** [rhuss/jolokia#775] [General]
Hi there, I am new to Jolokia. I am trying to deploy Jolokia to Tomcat webapps directory. I deployed jolokia-agent-war-unsecured-2.1.1.war to Tomcat 9.0.88 on Linux and to Tomcat 9.0.78 on Windows. The war is extracted / deployed in the webapps directory but when I access the web interface the result is always "HTTP Status 404 – Not Found". In the Tomcat log I don' see error messages, only a message that it is deployed.

Per my understanding deployed of the "unsecure version should be as simple as dropping the war file in the webapps directory but this is not the case? There seems to be restrictions when it comes to deploying to Tomcat? What am I missing?

https://repo1.maven.org/maven2/org/jolokia/jolokia-agent-war-unsecured/2.1.1/jolokia-agent-war-unsecured-2.1.1.war
https://repo1.maven.org/maven2/org/jolokia/jolokia-agent-war/2.1.2/jolokia-agent-war-2.1.2.war

**Comment by grgrzybek** [comment]
Hello

Mind that Tomcat 9 uses Servlet API 4 (`javax.servlet` packages), while Tomcat 10 uses Servlet API 5+ (`jakarta.servlet` packages).

Jolokia 2 is based on the latter, so please try with Tomcat 10+.
